flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/FileSystemTableSource.java

```diff
@@ -269,10 +271,35 @@ private SourceProvider createSourceProvider(BulkFormat<RowData, FileSourceSplit>
         tableOptions
                 .getOptional(FileSystemConnectorOptions.SOURCE_MONITOR_INTERVAL)
                 .ifPresent(fileSourceBuilder::monitorContinuously);
+        tableOptions
+                .getOptional(FileSystemConnectorOptions.SOURCE_REGEX_PATTERN)
+                .ifPresent(
+                        s -> {
+                            String regexPath = connectBasePathAndRegex(path.getPath(), s);
+                            fileSourceBuilder.setFileEnumerator(
+                                    bulkFormat.isSplittable()
+                                            ? () -> new BlockSplittingRegexEnumerator(regexPath)
+                                            : () -> new NonSplittingRegexEnumerator(regexPath));
+                        });
 
         return SourceProvider.of(fileSourceBuilder.build());
     }
 
+    private String connectBasePathAndRegex(String basePath, String regex) {
+        StringBuilder result = new StringBuilder();
+        result.append(basePath);
+        if (!basePath.endsWith(Path.SEPARATOR)) {
+            result.append(Path.SEPARATOR);
+        }
+        int startIndex = 0;
+        while (startIndex < regex.length()
```

Comment:
I think this loop is removing multiple path separators at the start of regex, for example `////.*`
Can there be a regex like this `////.*///.*`

Reply:
Still, I'm try to understand why need `connectBasePathAndRegex`. It turns out that what the option `source.regex-pattern` is for?
 From the source code, seems it'll look like if the table path is `/dir`,  the `source.regex-pattern` is `t/*`, it will try to match `/dir/t/*`?

TBH, I'm not sure it's reasonable or not for it look wired or unclear to me.
Another choice is user need to specifc `/dir/t/*` in `source.regex-pattern`.

I don't have a strong perference about these two choice, and   don't know which is a better choice. Let's hear other's voice.

Reply:
@luoyuxia Using a base path will help to reduce the search time. 
If we only provide `source.regex-pattern` , we have to search files from root path `/` when `source.regex-pattern` = `'/dir/t/.*/inner/.*'`.
If we provide base dir and `source.regex-pattern`, we will start to search files under the base dir `/dir/t/` when base dir = `'/dir/t'` and `source.regex-pattern` = `'.*/inner/.*'`.

Reply:
No, I mean user provide base dir and `source.regex-pattern`. But the `source.regex-pattern` should also be absolute path insteading of relative path.  But we only serach from the base dir  and to check each file.

Take this for example, `source.regex-pattern` will be `/dir/t/.*/inner/.*` which looks more clear than `.*/inner/.*` to me.
I'm wondering whether use `/dir/t/.*/inner/.*` will be more clear.